pkg/controllers/counter/controller.go

```diff
@@ -39,46 +42,40 @@ import (
 	"github.com/aws/karpenter-core/pkg/utils/resources"
 )
 
-var _ corecontroller.TypedController[*v1alpha5.Provisioner] = (*Controller)(nil)
-
 // Controller for the resource
 type Controller struct {
 	kubeClient client.Client
 	cluster    *state.Cluster
 }
 
 // NewController is a constructor
-func NewController(kubeClient client.Client, cluster *state.Cluster) corecontroller.Controller {
```

Comment:
What do you think about combining this controller with the provisioner controller?

Reply:
We _should_ do that but I think that change should be in a separate PR